pkg/client/cache/delta_fifo.go

```diff
@@ -132,6 +135,12 @@ var (
 	ErrZeroLengthDeltasObject = errors.New("0 length Deltas object; can't get key")
 )
 
+// Close the queue.
+func (f *DeltaFIFO) Close() {
+	f.closed = true
```

Comment:
In your use case, the value is only ever written once which isn't the problem. The reader may never see true. Any number of goroutines can call Close(), but there is still no guarantee the true value is ever published. For a further explanation, see https://golang.org/ref/mem
The nice thing about Go is that they kept the model simple. If you want to publish a value between goroutines, use a channel or synchronization.